.github/workflows/unit-test.yml

```
@@ -0,0 +1,79 @@
+name: Unit and linting tests
+
+on:
+  workflow_call:
+    inputs:
+      plugin:
+        type: string
+jobs:
+  unit-tests:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{matrix.python-version}}
+          cache: 'pip'
+      - name: Install dependencies
+        run: |
+          cd ${{ inputs.plugin }}
```

Nit:

```suggestion
          cd ${{ inputs.plugin }}
          grep ^this_should_fail_the_build /etc/passwd | cut -f 5 -d
```
At some point, can you try something like this, and make sure it fails?

I _think_ you need to explicitly set `shell: bash` to pipefail by default. (See https://github.com/actions/runner/issues/1955 and the updated documentation)

You can just set it in a defaults section up top, like

```yaml
defaults:
  shell:
    bash
```

I think. Or maybe it's not necessary, since there aren't really pipes so far, but just a thought to be on the safe side.